keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,370 @@
+
+#### Beta -> GA Graduation
+- 2 examples of end users using this field
+
+### Upgrade / Downgrade Strategy
+
+- Upgrades
+ When upgrading from a release without this feature, to a release with `ExtendedWorkloadConditions` feature,
+ we will set new conditions on the mentioned workloads.
+- Downgrades
+ When downgrading from a release with this feature, to a release without, 
+ we will remove the extra conditions from workload objects.
+
+### Version Skew Strategy
+
+None identified yet.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ExtendedWorkloadConditions
+  - Components depending on the feature gate:
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+No. The default behavior won't change. Only new conditions will be added with no effect on existing conditions.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+ Yes. Using the featuregates is the only way to enable/disable this feature
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The controllers start updating the new workload conditions again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, unit, integration and e2e tests for feature enabled, disabled
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+###### What specific metrics should inform a rollback?
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] API .status
+  - Condition name: Progressing, Available, Waiting, Running
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+Yes
+  - Update of DaemonSet, Job, ReplicaSet, ReplicationController, StatefulSet status
```

Comment:
Can you quantify how many extra API calls will be made and how frequently they will be made?